src/com/android/settings/security/AutoRebootPreferenceController.java

```diff
@@ -0,0 +1,109 @@
+    private static final String VALUE_STORE_KEY = "settings_reboot_after_timeout";
+    private static final boolean DEBUG = flase;
+
+    private PreferenceCategory mSecurityCategory;
+    private boolean mIsAdmin;
+    private final UserManager mUm;
+
+    public AutoRebootPreferenceController(Context context) {
+        super(context);
+        mUm = UserManager.get(context);
+    }
+
+    @Override
+    public void displayPreference(PreferenceScreen screen) {
+        super.displayPreference(screen);
+        mSecurityCategory = screen.findPreference(PREF_KEY_SECURITY_CATEGORY);
+        updatePreferenceState();
+    }
+
+    @Override
+    public boolean isAvailable() {
+        mIsAdmin = mUm.isAdminUser();
+        return mIsAdmin;
+    }
+
+    @Override
+    public String getPreferenceKey() {
+        return KEY_AUTO_REBOOT;
+    }
+
+    // TODO: should we use onCreatePreferences() instead?
+    private void updatePreferenceState() {
+
+        if (mSecurityCategory == null) {
+            log("updatePreferenceState called with null mSecurityCategory");
+            return;
+        }
+
+        if (mIsAdmin) {
+            ListPreference denyNewUsb = (ListPreference) mSecurityCategory.findPreference(KEY_AUTO_REBOOT);
+            log("updatePreferenceState called from admin/owner user update psfs value with currentValue :  " + currentValueInHours());
+            if(denyNewUsb != null){
+                denyNewUsb.setValue(currentValueInHours());
+            }
+        } else {
+            log("updatePreferenceState isn't called from admin/owner user removing ");
+            mSecurityCategory.removePreference(mSecurityCategory.findPreference(KEY_AUTO_REBOOT));
+        }
+    }
+
+    @Override
+    public void onResume() {
+        updatePreferenceState();
+    }
+
+    @Override
+    public boolean onPreferenceChange(Preference preference, Object value) {
+        final String key = preference.getKey();
+        log("onPreferenceChange key : " + key + " value : " + value + " isString? : " + (value instanceof  String) + " isAdminUser? : " + mUm.isAdminUser());
+        if (KEY_AUTO_REBOOT.equals(key) && value instanceof  String && mUm.isAdminUser() ) {
+            int timeout = Integer.parseInt((String) value);
+            Settings.Global.putInt(mContext.getContentResolver(), VALUE_STORE_KEY, hoursToMilli(timeout));
+        }
+        return true;
+    }
+
+    private String currentValueInHours() {
+        int value = Settings.Global.getInt(mContext.getContentResolver(), VALUE_STORE_KEY, 0);
+        return String.valueOf(milliToHours(value));
+    }
+
+    private int milliToHours(int milli){
+        //1 h = 3600000 ms ( ((((hour * 60) min) * 60) sec) * 1000 milli)
+        return milli / 3600000;
+    }
+
+    private int hoursToMilli(int hour){
+        //1 h = 3600000 ms ( ((((hour * 60) min) * 60) sec) * 1000 milli)
+        return hour * 3600000;
+    }
```

Comment:
I think using `java.util.concurrent.TimeUnit` with `TimeUnit.HOURS.toMillis(long)` and `TimeUnit.MILLISECONDS.toHours(long)` would be better

Reply:
TimeUnit work better with Long and not Integer, of course converting long into int and int into long is not a big deal, but it does not offer any meaningful changes.

Reply:
I am thinking we should just be using longs for time-related values anyway in order to futureproof this implementation (though the current max is 24.8551348 days from `Integer.MAX_VALUE`)

Reply:
[done](https://github.com/GrapheneOS/platform_packages_apps_Settings/pull/38/commits/a98f302dfef66cdd046092c26e680df9920da52b). 


cc @anupritaisno1 
Implementation has been changed from int to long.
So use `Settings.Global.getLong()` instead of `Settings.Global.getInt()` .

I have not tested it's, Simply because this is slight changes. When i do i will let you know you.

Reply:
I am marking this as resolved, Feel free open it again if you think otherwise .